include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -7082,7 +7082,7 @@ namespace VectorTools
         double my_values[3] = { mean_double.real(), mean_double.imag(), area };
         double global_values[3];
 
-        MPI_Allreduce (&my_values, &global_values, 3, MPI_DOUBLE,
+        MPI_Allreduce (my_values, global_values, 3, MPI_DOUBLE,
```

Comment:
Huh, interesting. I stand corrected, the code really did behave correctly. (I had to try it out...)

Anyways, this PR is still useful because it removes warnings.